library/Freckle/Test/Yesod/StatusAssertions.hs

```diff
@@ -0,0 +1,167 @@
+module Freckle.Test.Yesod.StatusAssertions
+  ( statusIs
+
+    -- * Expect success
+  , getOk
+  , postOk
+  , patchOk
+  , deleteOk
+
+    -- * Expect error
+  , getErr
+  , postErr
+  , patchErr
+  , deleteErr
+  )
```

Comment:
I personally don't like these, they seem like a low-value combinator that is too inflexible to be useful and just invites frustration. What if I'm using `getOk` and suddenly I need params? Where is `putOk`? What if I need headers on my `postErr`? Are we heading towards every `{method}(Ok|Err)(With(Body)(And)(Params))` combination? _We had something for that, it was called `request`. They've played us for absolute..._ anyway.

I look at something like,

```hs
getOk MyRoute
bodyDeepPermutationsAsJson whatever
```

And I just think doing the equivalent thing more directly is not much more effort, and the test is more explicit and extensible. I'm also a big fan of clearly-3-phase tests and `getOk` is mixing act/assert.

```hs
get MyRoute

statusIs 200
body <- getJsonBody
body `shouldMatchJson` whatever
```

Not really pushing back, but just offering a take in case you agree.